arangod/RocksDBEngine/RocksDBMetadata.cpp

```diff
@@ -94,11 +95,13 @@ RocksDBMetadata::RocksDBMetadata()
  * @param  seq   The sequence number immediately prior to call
  * @return       May return error if we fail to allocate and place blocker
  */
-Result RocksDBMetadata::placeBlocker(TransactionId trxId, rocksdb::SequenceNumber seq) {
+Result RocksDBMetadata::placeBlocker(TransactionId trxId, rocksdb::SequenceNumber& seq) {
   return basics::catchToResult([&]() -> Result {
     Result res;
     WRITE_LOCKER(locker, _blockerLock);
 
+    seq = std::max(seq, _maxBlockersSequenceNumber);
```

Comment:
For each collection, we keep track of the highest assigned sequence number used for a blocker.

This is necessary because otherwise readers could be reading the list of blockers sequence numbers and determine its lower bound, assuming that no lower seqnos will ever be added later. That is at least an assumption that readers currently make. This can happen in rare cases when two write operations were in flight at the same time but scheduled in adverse order. In this case, the higher seqno could be inserted first, and then the lower one. If in-between a reader would have fetched the lower bound, it was not aware of the in-flight operation which would insert a lower value later.

With this change, this is now prevented.

Reply:
So it would in principle suffice to use the minimum sequence number of any blocker, correct? And I suppose you used the maximum instead because it's slightly cheaper to keep up to date?